app/Nova/Metrics/ActiveAttendanceBreakdown.php

```diff
@@ -27,20 +33,42 @@ public function name(): string
      */
     protected $showAllTime = false;
 
+    /**
+     * If displaying on the main dashboard, this indicates the event to get attendance from.
+     */
+    private int $resourceId = -1;
+
+    /**
+     * If displaying on a page different from the type of attendable you are showing.
+     * This indicates the morphClass of the attendable (for example, an event).
+     */
+    private string $attendableType;
+
     /**
      * Create a new ActiveAttendanceBreakdown metric.
      */
-    public function __construct(bool $showAllTime = false)
+    public function __construct(bool $showAllTime = false, int $resourceId = -1, string $attendableType = '')
     {
         parent::__construct();
+        $this->resourceId = $resourceId;
         $this->showAllTime = $showAllTime;
+        $this->attendableType = $attendableType;
     }
 
     /**
      * Calculate the value of the metric.
      */
     public function calculate(Request $request): PartitionResult
     {
+        $resourceId = -1;
+        $attach_to_resource_id = false;
+        if ($this->resourceId !== -1) {
+            $resourceId = $this->resourceId;
+            $attach_to_resource_id = true;
+        } elseif (isset($request->resourceId)) {
+            $resourceId = $request->resourceId;
+            $attach_to_resource_id = true;
+        }
```

Comment:
```suggestion
        $resourceId = $request->resourceId ?? $this->resourceId;
```

Reply:
The issue with this suggestion is that the ActiveAttendanceBreakdown doesn't always attach to a resource ID. I attempted to do the resource ID things in the way you suggested, and it broke the card for me.

Reply:
Not working for now - Will revisit after trying a few other cards